Marlin/ultralcd.cpp

```diff
@@ -193,6 +193,7 @@ uint16_t max_display_update_time = 0;
   #endif
 
   #if ENABLED(ADVANCED_PAUSE_FEATURE)
+    void lcd_change_filament_menu();
     void lcd_advanced_pause_toocold_menu();
```

Comment:
@tcm0116 — This menu has to stay because it is used below.

Reply:
I'm eliminating `lcd_enqueue_filament_change` as it is not really necessary.

Instead of checking `thermalManager.tooColdToExtrude()` when enabling the Change Filament menu options, it'll check `thermalManager.targetTooColdToExtrude()`, which will ensure that a reasonable target temperature has been set for the extruder. `ensure_safe_temperature` will then do it's job and make sure that the temperatures are good to go before performing the change. I did find that there was an issue where the nozzle wouldn't park while it was waiting in `ensure_safe_temperature`, so I re-arranged the sequence to ensure that the nozzle parks before calling `ensure_safe_temperature`. This works nicely with the changes from #8836 that were incorporated by skipping the retract if the nozzle is too cold, parking, waiting for the extruder to heat, and then performing the change.